### DISCUSSION TITLE
Format of "available" apis (ip-ranges/int:pk/available-ips/)

### DISCUSSION THREAD

**Discussion by arthanson** [netbox-community/netbox-python#10] [Polls]
ip-ranges, prefixes and vlan-groups have sub-api's that are currently not modeled in netbox-python, opening a discussion of what format should be used.  Here are the different formats that I could think of:

1. nb.ipam.ip_range.available_ips.list(1)
2. nb.ipam.ip_ranges(1).available_ips.list()
3. IPRange.get_available_ips()
4. nb.ipam.available_ips(ip_range=1).list()

Option 1 is the most in-line with the way the classes are currently, but option 2 seems more in-line with pynetbox. 

These would be for the following endpoints:
        'ip-ranges/<int:pk>/available-ips/',
        'prefixes/<int:pk>/available-prefixes/',
        'prefixes/<int:pk>/available-ips/',
        'vlan-groups/<int:pk>/available-vlans/',

**Comment by markkuleinio** [comment]
I only now realized this repo, nice! I guess it's no use submitting PRs anymore to pynetbox 😁

**Comment by pobradovic08** [comment]
Option 2 is maybe slightly more intuitive as the ID is related to `ip_range` and not `available_ips`, but I voted for Option 1 as it could be easily implemented. I don't know how much work is needed to implement Option 2 and if that would be bring huge benefits compared to Option 1.
